src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -208,13 +219,23 @@ export const useSwapper = () => {
           action,
         })
 
-        const tradeQuote = await swapper.getTradeQuote({
-          sellAsset,
-          buyAsset,
-          sellAmount,
-          sendMax: false,
-          sellAssetAccountNumber: 0,
-        })
+        const tradeQuote: TradeQuote<KnownChainIds> = await (async () => {
+          if (sellAsset.chainId === 'eip155:1') {
+            return swapper.getTradeQuote({
+              chainId: 'eip155:1',
+              sellAsset,
+              buyAsset,
+              sellAmount,
+              sendMax: false,
+              sellAssetAccountNumber: 0,
+            })
+          } else if ('bip122:000000000019d6689c085ae165831e93') {
+            // TODO do bitcoin specific trade quote including `bip44Params`, `accountType` and `wallet`
+            // They will need to have selected an accountType from a modal if bitcoin
+            throw new Error('bitcoin unsupported')
+          }
+          throw new Error(`unsupported chain id ${sellAsset.chainId}`)
```

Comment:
similarly here to @cjthompson comment, can you please try/catch everything inside this IIFE?

Reply:
Wouldn't the error just bubble up to the calling function where there is a try/catch?